### PR TITLE
Cleaned some clothing flags.

### LINKED ISSUE #8458: Ninja shoes cover eyes?
I have no idea what is going on here, but when doing surgery on someone's eyes, if they are wearing the ninja suit shoes, it will say that their eyes need to be exposed first.

Steps to reproduce:
Strip completely
spawn and equip a pair of ninja suit shoes
use the surgical drapes to start an eye surgery
be confused as to why it says the eyes are covered.

### PR DESCRIPTION
**Pull request by xxalpha**
~~Port of https://github.com/d3athrow/vgstation13/pull/5099~~
edit: Optimization code not worth it.

Plus some necessary flag cleaning.

Fixes #8458

### CODE REVIEW

**Comment by RemieRichards** [comment]
I saw this PR on vg.
I did not like it.

I don't like the concept of having to give moving things a flag for other things to then check that they even moved. I don't feel it's worth it, and that it would end up causing situations where a flag is missing and odd behaviour resulting.

**Comment by Iamgoofball** [comment]
@RemieRichards But it's better optimized.

**Comment by RemieRichards** [comment]
@Iamgoofball ```HasProximity()``` isn't even that laggy of a proc, I have no idea why this was something VG targeted as an optimisation.

**Comment by Iamgoofball** [comment]
@RemieRichards but it is used in turf/Entered() which is heavily used

so cutting down on all lag from Entered() is a good thing

**Comment by xxalpha** [comment]
Yeah entered is constantly used and it's calling range() everytime anything moves.

**Comment by RemieRichards** [comment]
@Iamgoofball Again, I don't care for this method of doing it, I will never care for this method of doing it.

**Comment by Midaychi** [comment]
Tested this branch on wine under linux, pile of active flashers and prox sensors, total cpu use on a 1.4ghz celery stick version of a haswell was between 6 and 14 according to the gc

**Comment by Fox-McCloud** [comment]
What does cleaning up clothing flags have to do with fixing up the entered proc? Aren't PR's meant to be highly atomic?

**Comment by Midaychi** [comment]
It's because in tg uses a ton of flags and they ran into a limit, so in order to implement this they had to clean up some flags elsewhere

**Comment by optimumtact** [comment]
can we see some profiling stats before we bother with this?

I prefer code cleanliness unless the profiler shows significant change.

**Comment by ComicIronic** [comment]
>not even that expensive

enteredold
It ended up as our fourth most costly proc in terms of self. recursive_hear_check only had slightly higher, but for 29 times the number of calls.

As for the performance difference, I made a post on the vg PR about it.
It's a pretty good drop.

**Comment by xxalpha** [comment]
I'll run a test when I get home as well.

**Comment by xxalpha** [comment]
Singularity test (from the engine to the armory)
Without optimization:
singularitytest_nooptimization

With optimization code:
singularitytest_withoptimization

**Comment by RemieRichards** [comment]
those numbers.. aren't that great at all.

**Comment by xxalpha** [comment]
Yep

**Comment by xxalpha** [comment]
Do you want to at least salvage the flag cleaning?

**Comment by RemieRichards** [comment]
@xxalpha the flag stuff is great yeah.

**Comment by MrStonedOne** [comment]
pre-op
/turf/entered 0.000212292103515192560659538166994
/turf/space/entered 0.000222579871930965996644180392425
post-op
/turf/entered 0.000178633670268386197281282676891
/turf/space/entered 0.000215070582254721776874046846283

**Comment by xxalpha** [comment]
It really shows improvement when running through a corridor of portable flashers, but...that's not a real case scenario.

**Comment by RemieRichards** [comment]
Yeah those numbers aren't worth shitting up the code.

**Comment by xxalpha** [comment]
This is now just some clothing flags clean up.

**Comment by Fox-McCloud** [comment]
Still don't see the point of this PR, to be honest.

**Comment by optimumtact** [comment]
We're running out of flag space

**Comment by RemieRichards** [comment]
@Fox-McCloud bitfields can only contain at maximum 16 bitflags, our ```flags``` bitfield has been 3 flags over the limit (that's 3 flags that share numbers with other flags causing weirdness like ninja shoes covering the eyes for surgery) for a LONG ASS time.

we use a bitfield instead of a list as checking if a bit is active in a bit field is MUCH cheaper than checking if an element is in a list.

**Comment by ComicIronic** [comment]
>It really shows improvement when running through a corridor of portable flashers, but...that's not a real case scenario.

What? No, it shows less improvement while running through a corridor of portable flashers, because the average Entered ends up costing more. It still shows a good bit of improvement.

Fundamentally, I think your Entered is just less costly than ours, so you never had a reason to port it in the first place.

**Comment by Fox-McCloud** [comment]
@RemieRichards 

Thanks for the clarification; I'll have to keep an eye on this myself, as our flags are just about as bad.

**Comment by RemieRichards** [comment]
@ComicIronic it's probably because you guys ```spawn()``` off the ```HasProximity()``` calls for simultaneousness, simultaneousness is 90% of the reason explosions lag, because it all happens simultaneously

File: code/__DEFINES/flags.dm [hunk 1]
```diff
@@ -19,12 +19,6 @@
 #define ON_BORDER		512		// item has priority to check when entering or leaving
 
 
-#define GLASSESCOVERSEYES	1024
-#define MASKCOVERSEYES		1024		// get rid of some of the other retardation in these flags
-#define HEADCOVERSEYES		1024		// feel free to realloc these numbers for other purposes
-#define MASKCOVERSMOUTH		2048		// on other items, these are just for mask/head
-#define HEADCOVERSMOUTH		2048
-
```
Comment (phil235) [inline code/__DEFINES/flags.dm]: Since you removed those, you could lower the number used for headbangprotect and probably other flags.

Comment (xxalpha) [inline code/__DEFINES/flags.dm]: There are still conflicts.

**Pull request merged**